pylint/checkers/utils.py

```diff
@@ -1919,24 +1919,29 @@ def in_type_checking_block(node: nodes.NodeNG) -> bool:
     return False
 
 
-def is_typing_literal(node: nodes.NodeNG) -> bool:
-    """Check if a node refers to typing.Literal."""
+def is_typing_member(node: nodes.NodeNG, typing_members: str | tuple[str, ...]) -> bool:
+    """Check if `node` is a member of the `typing` module and has one of the names from
+    `typing_members`.
+    """
     if isinstance(node, nodes.Name):
         try:
             import_from = node.lookup(node.name)[1][0]
         except IndexError:
             return False
+
+        if isinstance(typing_members, str):
+            typing_members = (typing_members,)
         if isinstance(import_from, nodes.ImportFrom):
-            return (  # type: ignore[no-any-return]
+            return (
                 import_from.modname == "typing"
-                and import_from.real_name(node.name) == "Literal"
+                and import_from.real_name(node.name) in typing_members
```

Comment:
Any reason not to use `qname()` here? We could do something with a for loop or comprehension and an f-string like `f'typing.{n}'`

Reply:
The existing function logic just worked the way I needed it to so I didn't feel it needed changing.
Do you mean use `qname` instead of `real_name`? It looks like [`real_name`](https://github.com/PyCQA/astroid/blob/main/astroid/nodes/_base_nodes.py#L153) is intended for the current use case?

Reply:
Never mind, that's indeed the correct method. Got my methods confused here, sorry!

Reply:
No worries at all; I like your analytical reviews!